### PR TITLE
feat(sdk): website api custom error page

### PR DESCRIPTION
**Pull request by meirdev**
Hi,

The PR allows you to set a custom error page in `cloud.Website` and `ex.React`.

Fixes the following issues:
#5118 
#4931

### CODE REVIEW

File: libs/wingsdk/src/target-tf-aws/react-app.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export class ReactApp extends ex.ReactApp implements aws.IAwsReactApp {
       {
         ...this._hostProps,
         path: this._buildPath,
+        errorDocument: "index.html",
```
Comment (tsuf239) [inline libs/wingsdk/src/target-tf-aws/react-app.ts:47]: can you please share the constant you used before?

**Review by tsuf239 — COMMENTED** [review]
Nice! can you please add tests written in wing to the following files: 
`examples/tests/sdk_tests/website/website.test.w` and `examples/tests/sdk_tests/website/react-app.test.w`

**Review by tsuf239 — COMMENTED** [review]
Looks Good!
I am running automated tests on the tf-aws platform: https://github.com/winglang/wing/actions/runs/7378135823

**Comment by tsuf239** [comment]
it seems like some of the PR tests failed, do you need any help debugging those?

**Comment by meirdev** [comment]
@tsuf239 It's weird, when I add the test in the end of the file:

```
test "redirect to index.html" {
  assert(http.get(website.url + "/page123").body.contains("Hello Wing!"));
}
```

The test failed in the previous test: `website is working`.

When I added debug prints (the first is `wingEnv` and the second is `expected`) I saw this:

```
fail ┌ react-app.test.wsim » root/env0/test:website is working    
     │ // This file is generated by wing
window.wingEnv = {
  "apiUrl": "http://127.0.0.1:63376",
  "anotherEnvVar": "preflight variable"
};
     │ // This file is generated by wing
window.wingEnv = {
  "apiUrl": "http://127.0.0.1:63374",
  "anotherEnvVar": "preflight variable"
};
```

Somehow adding a test in the same file causes the address to change.

**Comment by tsuf239** [comment]
you can maybe try to put this asssertion: `  assert(http.get(website.url + "/page123").body.contains("Hello Wing!"));
` inside of the "website is working" test? 
Also if you can open an issue for that bug it will be great :)

**Comment by tsuf239** [comment]
Looks good! 
waiting for this: https://github.com/winglang/wing/actions/runs/7407686934 to pass :)

**Review by tsuf239 — APPROVED** [review]

**Pull request merged**